network/p2p/router_test.go

```diff
@@ -0,0 +1,425 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p/mocks"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+func TestAppRequestResponse(t *testing.T) {
+	handlerID := uint64(0x0)
+	request := []byte("request")
+	response := []byte("response")
+	nodeID := ids.GenerateTestNodeID()
+	chainID := ids.GenerateTestID()
+
+	tests := []struct {
+		name        string
+		requestFunc func(t *testing.T, router *Router, client *Client, sender *common.MockSender, handler *mocks.MockHandler, wg *sync.WaitGroup)
+	}{
+		{
+			name: "app request",
+			requestFunc: func(t *testing.T, router *Router, client *Client, sender *common.MockSender, handler *mocks.MockHandler, wg *sync.WaitGroup) {
+				sender.EXPECT().SendAppRequest(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+					Do(func(ctx context.Context, nodeIDs set.Set[ids.NodeID], requestID uint32, request []byte) {
+						for range nodeIDs {
```

Comment:
should this be `for nodeID := range nodeIDs {`? seems `nodeID` will always be the static one defined above.
(note if we do this we should add `nodeID := nodeID` to the loop body as well to avoid capturing the loop var)
(similar to what we have in "app gossip specific")
comment applies a few places in this file.